### DISCUSSION TITLE
Lint and script checks fail

### DISCUSSION THREAD

**Discussion by rishujam** [oppia/oppia-android#4726] [Q&A (Other)] [answered]
I am contributing for the first time and I am facing lint errors in the code review. 

Lint errors: Import should be in lexicographic order 
but they are already in order (done using optimize import button)

Script check: Missing Kdoc for a file that I created, The file created does not have a corresponding test file.
So do I need to write a test too for each file that I add?

**Comment by rishujam** [comment]
found about lint: https://www.youtube.com/watch?v=yUno_88kKvU&ab_channel=MobileAcademy.

**Reply by BenHenning** [reply]
Thanks for following up @rishujam. Could you maybe update your answer with the specific steps you took to fix your issue? That could help other contributors who run into the same problem.

Also, it would be helpful if you could clarify in your question what the specific errors were that you encountered (as this can help contributors find this discussion).

**Comment by rishujam** [comment]
@BenHenning 

> I am contributing for the first time and I am facing lint errors in the code review.
> 
> Lint errors: Import should be in lexicographic order but they are already in order (done using optimize import button)
> 
> Script check: Missing Kdoc for a file that I created, The file created does not have a corresponding test file. So do I need to write a test too for each file that I add?

**Comment by BenHenning** [comment, marked as answer]
Thanks for updating the description @rishujam.

For each of the issues you're running into:
- Lexicographic order: is this a ktlint error? If so, Android Studio's optimize imports may not provide the correct ordering required by ktlint. You should make sure you've run the setup script per the installation instructions listed on our wiki: https://github.com/oppia/oppia-android/wiki/Contributing-to-Oppia-android#install-oppia-android. That will check ktlint for each push to GitHub that you do, and when ktlint fails it'll provide a command you can copy-and-run to auto-fix most style issues.
- Missing KDocs: this means you should add these. See: https://github.com/oppia/oppia-android/wiki/Static-Analysis-Checks#kdoc-validity-check.
- Missing test file: all new files do need tests (generally, sometimes there are reasonable exceptions such as: view models, presenters, and listeners). See: https://github.com/oppia/oppia-android/wiki/Static-Analysis-Checks#test-file-presence-check.

**Comment by rishujam** [comment]
![Screenshot 2022-11-27 235433](https://user-images.githubusercontent.com/74773876/204152937-5887ef07-9b08-4d57-85d2-e82dc1a47644.png)
Where can I run the lint command to fix the issue?
@BenHenning

**Reply by BenHenning** [reply]
@rishujam can you create a new discussion for this? I'm also a bit confused about your question: are you asking why the command isn't running correctly, or where to find the command? (Note that if it's the latter, the command is available as output from the ktlint_lint_check.sh script, as indicated in your screenshot).
